### DISCUSSION TITLE
How does `identify` determine the webp image format?

### DISCUSSION THREAD

**Discussion by romanb089** [ImageMagick/ImageMagick#4686] [General] [answered]
Software used:
```
~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal

~$ identify -version
Version: ImageMagick 7.1.0-18 Q16-HDRI x86_64 2021-12-04 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in): bzlib djvu fontconfig freetype jbig jng jpeg lcms lqr lzma openexr png tiff webp x xml zlib
Compiler: gcc (9.3)
```
I found that some graphics editors use the RIFF container for PNG images, so
identify identifies images such as WEBP:
```
identify  kormushka_medved_vid_sverhu_2_2.png
kormushka_medved_vid_sverhu_2_2.png WEBP 2000x2000 2000x2000+0+0 8-bit sRGB 32321B 0.070u 0:00.064
```
Sample image can be downloaded from the [link](https://bugs.chromium.org/p/webp/issues/attachment?aid=535218&signed_aid=jz7N0AQ8RqyGCPUWFJskXw==) (The original image was created using the program `SOLIDWORKS`)

Could you elaborate on exactly how the format identification takes place? 
I also opened a discussion with the WEBP developers: https://bugs.chromium.org/p/webp/issues/detail?id=547

**Comment by snibgo** [comment, marked as answer]
pngcheck also says that is not a valid PNG file:
```
f:\web\im>pngcheck kormushka_medved_vid_sverhu_2.png
kormushka_medved_vid_sverhu_2.png  this is neither a PNG or JNG image nor a MNG stream
ERROR: kormushka_medved_vid_sverhu_2.png
```

**Reply by romanb089** [reply]
Indeed, `pngcheck` returns an error. Thank you

**Comment by fmw42** [comment]
ImageMagick is generally very strict in conforming to any file format specifications. So if some argument in the header is missing or malformed, IM will complain.  

I am not sure if webp is any different, but IM finds the file type from the "magic" entry in the file header that identifies the file format rather than using the suffix. So if you use a different suffix, IM will still know what the file type is. If your file is png for example but has a .webp suffix, IM will identify it png, if the header is proper and the "magic" identifier says PNG. See %m at https://imagemagick.org/script/escape.php
